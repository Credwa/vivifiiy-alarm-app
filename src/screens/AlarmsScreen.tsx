import Background from '@/components/Background';
import * as React from 'react';
import { Pressable, StyleSheet, View } from 'react-native';
import VivText from '@/components/VivText';
import { AntDesign } from '@expo/vector-icons';
import { SafeAreaView } from 'react-native-safe-area-context';
import Colors from '@/constants/Colors';

export default function AlarmsScreen() {
  const createNewAlarm = () => {};

  return (
    <Background>
      <SafeAreaView style={styles.container}>
        <View style={styles.header}>
          <VivText fontName="Title4" color={Colors.greyLight1}>
            Alarms
          </VivText>
          <Pressable onPress={createNewAlarm} style={styles.addAlarm}>
            {({ pressed }) => (
              <AntDesign name="plus" size={28} color={pressed ? Colors.greyLight3 : Colors.greyLight1} />
            )}
          </Pressable>
        </View>
      </SafeAreaView>
    </Background>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    marginTop: 30
  },
  header: {
    flexDirection: 'row',
    flex: 1,
    justifyContent: 'space-between',
    alignContent: 'space-between'
  },
  addAlarm: {
    height: 44,
    width: 44,
    alignContent: 'center',
    alignItems: 'center',
    justifyContent: 'center'
  }
});
