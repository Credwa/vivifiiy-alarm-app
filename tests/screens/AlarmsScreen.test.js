import AlarmsScreen from '@/screens/AlarmsScreen';
import { Ionicons } from '@expo/vector-icons';
import * as Font from 'expo-font';
import * as React from 'react';
import renderer from 'react-test-renderer';

describe('<AlarmsScreen />', () => {
  beforeEach(async () => {
    await Font.loadAsync({
      ...Ionicons.font
    });
  });

  it(`renders correctly`, () => {
    const tree = renderer.create(<AlarmsScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
