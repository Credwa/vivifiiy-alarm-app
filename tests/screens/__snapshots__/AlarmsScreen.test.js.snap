// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AlarmsScreen /> renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "flex": 1,
      "paddingHorizontal": 20,
      "resizeMode": "cover",
    }
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <RNCSafeAreaView
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      Alarms
    </Text>
    <View
      style={
        Object {
          "height": 1,
          "marginVertical": 30,
          "width": "80%",
        }
      }
    />
  </RNCSafeAreaView>
</View>
`;
